text/0000-layer-table.md

```diff
@@ -0,0 +1,89 @@
+# Meta
+[meta]: #meta
+- Name: Move layer types to new table in layer.toml
+- Start Date: 2021-01-05
+- Author(s): natalieparellano
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Introduces a new table in `<layers>/<layer>.toml` to enable Bash buildpacks to easily opt in to layer re-use.
+
+# Definitions
+[definitions]: #definitions
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 0052](https://github.com/buildpacks/rfcs/blob/main/text/0052-opt-in-layer-caching.md) described a change (to be implemented in Buildpack API 0.6) whereby buildpacks must opt in to layer re-use - i.e., each `<layers>/<layer>.toml` will be restored with no `launch`, `build`, or `cache` flags set and buildpacks must modify the restored TOML to set these flags.
+
+The RFC assumed that Bash buildpacks would be able to easily opt in to layer re-use by appending lines to the TOML, e.g.:
+
+`echo "launch = true" >> layer.toml`
+
+However, because `launch`, `build`, and `cache` are top level keys, the above command would add a `launch` key to the `[metadata]` table instead of the top level.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes moving the `launch`, `build`, and `cache` keys into a new `types` table in `<layers>/<layer>.toml`:
+
+```toml
+[types]
+launch = true
+build = false
+cache = false
+```
+
+A Bash buildpack could then write something like the following:
+
+```bash
+cat >> launch.toml <<EOL
+[types]
+launch = true
+EOL
+```
+
+This change should be implemented with Buildpack API 0.6 so that it goes hand-in-hand with the opt in layer re-use feature.
+
+- If a buildpack implementing Buildpack API less than 0.6 tries to write `<layers>/<layer>.toml` with a `[types]` table, the lifecycle should fail.
```

Comment:
A buildpack that implements Buildpack API less than 0.6 won't need to write `launch`, `build` and `cache` because [RFC0052](https://github.com/buildpacks/rfcs/blob/main/text/0052-opt-in-layer-caching.md) doesn't affect it, right?
If yes, will `lifecycle` fail if the buildpack tries to write these values at the top level or will it be just part of `metadata` and there is no problem with it?

Reply:
That's right, `<layers>/<layer>.toml` will be restored with the top level keys intact for older buildpack apis. If a buildpack were to append those keys to the file they would just be part of metadata.

Reply:
I'm not sure but maybe it's worth adding it to the RFC.

Reply:
Added in 2f2d27347272f6bcd538a71dd93c673c0ff2c69c